cmd/kcp-catalog/bind/catalogentry/bind.go

```diff
@@ -0,0 +1,249 @@
+				GenerateName: ref.Workspace.ExportName + "-",
+			},
+			Spec: apisv1alpha1.APIBindingSpec{
+				Reference: ref,
+			},
+		}
+
+		apiBindings = append(apiBindings, *apiBinding)
+	}
+
+	// Create bindings to the target workspace
+	for _, binding := range apiBindings {
+		found, err := bindingAlreadyExists(ctx, binding, kcpClient, b.Out)
+		if err != nil {
+			allErrors = append(allErrors, err)
+		}
+
+		if !found {
+			err := kcpClient.Create(ctx, &binding)
+			if err != nil {
+				allErrors = append(allErrors, err)
+			}
+		}
+
+	}
+
+	if err := wait.PollImmediate(time.Millisecond*500, b.BindWaitTimeout, func() (done bool, err error) {
+		availableBindings := []apisv1alpha1.APIBinding{}
+		for _, binding := range apiBindings {
+			createdBinding := apisv1alpha1.APIBinding{}
+			err = kcpClient.Get(ctx, types.NamespacedName{Name: binding.Name}, &createdBinding)
+			if err != nil {
+				return false, err
+			}
+			availableBindings = append(availableBindings, createdBinding)
+		}
+		return bindReady(availableBindings), nil
+	}); err != nil {
+		return fmt.Errorf("bindings for catalog entry %s could not be created successfully: %v", entryName, err)
+	}
+
+	if _, err := fmt.Fprintf(b.Out, "%s created and bound to catalog entry.\n", entryName); err != nil {
+		allErrors = append(allErrors, err)
+	}
+	return utilerrors.NewAggregate(allErrors)
+}
+
+func bindReady(bindings []apisv1alpha1.APIBinding) bool {
+	for _, binding := range bindings {
+		if binding.Status.Phase != apisv1alpha1.APIBindingPhaseBound {
+			return false
+		}
+	}
+	return true
+}
+
+func newClient(cfg *rest.Config, clusterName logicalcluster.Name) (client.Client, error) {
+	scheme := runtime.NewScheme()
+	err := apisv1alpha1.AddToScheme(scheme)
+	if err != nil {
+		return nil, err
+	}
+
+	err = catalogv1alpha1.AddToScheme(scheme)
+	if err != nil {
+		return nil, err
+	}
+	return client.New(kcpclienthelper.SetCluster(rest.CopyConfig(cfg), clusterName), client.Options{
+		Scheme: scheme,
+	})
+}
+
+// bindingAlreadyExists lists out the existing bindings in a workspace, checks if the export reference is the same. If so,
+// it further checks the permission claims and updates the existing binding's claims.
+func bindingAlreadyExists(ctx context.Context, expectedBinding apisv1alpha1.APIBinding, kcpclient client.Client, wr io.Writer) (bool, error) {
+	found := false
+	bindingList := apisv1alpha1.APIBindingList{}
+	err := kcpclient.List(ctx, &bindingList)
+	if err != nil {
+		return found, err
+	}
+
+	for _, b := range bindingList.Items {
+		if b.Spec.Reference == expectedBinding.Spec.Reference {
+			found = true
+			// if the specified export reference matches the expected export reference, then check if permission
+			// claims also match.
+			if !reflect.DeepEqual(b.Spec.PermissionClaims, expectedBinding.Spec.PermissionClaims) {
+				// if the permission claims are not equal then update the apibinding
+				b.Spec = expectedBinding.Spec
+				err := kcpclient.Update(ctx, &b)
```

Comment:
I would not recommend performing a mutation inside a function whose name connotes a read-only action.